### PR TITLE
#168131432 Add Search Functionality

### PR DESCRIPTION
**Pull request by willywax**
#### What does this PR do?
add request search functionality
#### Description of Task to be completed?
add ability for manager to search requests by
- `id` - which is the requestId
- `requester` - which can pass first name or last name of the requester
- `status` - which can be `pending, approved, rejected`
- `travelDate or returnDate` - requests with format `yyyy-mm-dd`
- `from` - represents the origin
- `reason` - any word in reason 
- `accommodation` - search by accommodation name
- `destination` - search by destination name

#### How should this be manually tested?
- clone the repo and checkout to `ft-search-requests-168131432` branch
- run `npm install` then `npm run test` to check it tests pass
- Login as a manager and head to postman on `api/v1/search/requests` endpoint  with GET method to get all requests.
- Alternatively you can search by any params as shown in the description 
`api/v1/search/requests?status=pending&requester=Jonathan`
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[ #168131432](https://www.pivotaltracker.com/story/show/168131432)
#### Screenshots (if appropriate)
image
image
image

#### Questions:
N/A

### CODE REVIEW

File: src/utils/search.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  searchData(data, filters) {
+    const results = [];
+    const filterKeys = Object.keys(filters);
+
+    if (filterKeys.length === 0) return data;
+
+    for (let count = 0; count < data.length; count++) {
+      let match = false;
+      for (const key of filterKeys) {
+        if (key === 'requester') {
```
Comment (Paulstar200) [inline src/utils/search.js]: Consider using a switch statement here

**Review by drayzii — COMMENTED** [review]
LGTM

**Review by drayzii — APPROVED** [review]

**Review by Dkabiswa — APPROVED** [review]

**Review by j0flintking02 — APPROVED** [review]
LGTM

**Review by Paulvitalis200 — APPROVED** [review]
Good work on this feature

**Pull request merged**